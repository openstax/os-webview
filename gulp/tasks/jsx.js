const path = require('path');
const gulp = require('gulp');
const config = require('../config');
const pi = require('gulp-load-plugins')({
    pattern: ['gulp-*', 'gulp.*', 'del']
});

function jsx() {
    const configSrcApp = `${config.src}/app`;
    const configDestApp = `${config.dest}/app`;

    return gulp.src(`${configSrcApp}/**/*.jsx`, {
        since: gulp.lastRun(jsx)
    })
    .pipe(pi.sourcemaps.init({loadMaps: true}))
    .pipe(pi.rename((uri) => {
        uri.extname = '.jsx.js';
    }))
    .pipe(pi.babel({
        compact: false,
        presets: ['@babel/preset-react']
    }))
    // prefix the sourcemaps with '../src/' so webpack can find them
    .pipe(pi.sourcemaps.mapSources(function(sourcePath, file) {
      const sourcePathHtml = sourcePath.replace('.jsx.js', '.jsx');
      const rel = path.relative(path.dirname(`${configDestApp}/${sourcePath}`), `${configSrcApp}/${sourcePathHtml}`)
      return rel;
    }))
    .pipe(pi.sourcemaps.write('.'))
    .pipe(gulp.dest(`${configDestApp}`));
}


function watchJsx() {
    gulp.watch([`${config.src}/**/*.jsx`], config.watchOpts, gulp.series(
        jsx
    ));
}

exports.jsx = jsx;
exports.jsx.watch = watchJsx;
