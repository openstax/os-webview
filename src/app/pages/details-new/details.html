<template args="model">
    <div class="hero" data-slug="{model.slug}">
        <div class="content book-title">
            <h1>{model.bookTitle}</h1>
        </div>
    </div>
    <div class="content-wrapper">
        <div class="content">
            <div class="phone-view" skip="true"></div>
            <div class="bigger-view">
                <div class="tab-controller" skip="true"></div>
                <div class="tab-content" skip="true"></div>
            </div>
        </div>
    </div>
</template>
