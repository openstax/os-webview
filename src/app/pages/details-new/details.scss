@import 'pattern-library/core/pattern-library/headers';

@mixin set-color-items($color) {
    @at-root .#{$color} {
        .hero,
        .book-icon,
        .webinars .icon-cell {
            background-color: os-color($color);
            color: text-color(white);
        }

        .hero {
            background-image: linear-gradient(to right, os-color($color), lighten(os-color($color), 7));
        }

        @if ($color == yellow) {
            .book-icon,
            .webinars .icon-cell {
                color: text-color(normal);
            }
        }
    }

    @at-root .fg-#{$color} {
        .hero {
            color: os-color($color);
        }
    }
}

.details-page-v2 {

    @include wider-than($phone-max) {
        background-color: ui-color(white);

        &.card-background {
            > .content-wrapper {
                background-image: linear-gradient(to bottom, ui-color(white), ui-color(white) 10rem, ui-color(default) 25rem);

                @include wider-than($tablet-max) {
                    background-image: linear-gradient(to bottom, ui-color(white), ui-color(white) 15rem, ui-color(default) 30rem);
                }
            }
        }
    }

    .hero {
        @include set-color-items(gold);
        @include set-color-items(green);
        @include set-color-items(blue);
        @include set-color-items(deep-green);
        @include set-color-items(gray);
        @include set-color-items(light-blue);
        @include set-color-items(orange);
        @include set-color-items(yellow);
        @include set-color-items(red);

        align-items: flex-end;
        background-image: linear-gradient(75deg, #aaa, #777);
        color: text-color(normal);
        display: flex;
        min-height: 16.5rem;
        overflow: hidden;
        padding-bottom: 5rem;

        @include width-between($phone-max, $tablet-max) {
            min-height: 27.3rem;
            padding-bottom: 6rem;
        }

        @include wider-than($tablet-max) {
            min-height: 34.3rem;
            padding-bottom: 7rem;
        }

        .book-title {
            padding: 0;

            h1 {
                @include title-font(4rem);

                text-align: left;
                // white-space: nowrap;

                @include width-between($phone-max, $tablet-max) {
                    @include title-font(10rem);
                }

                @include wider-than($tablet-max) {
                    @include title-font(13rem);
                }
            }
        }
    }

    .content {
        @extend %content;

        justify-content: flex-start;
    }

    .phone-view {
        display: block;

        @include wider-than($phone-max) {
            display: none;
        }
    }

    .bigger-view {
        display: none;

        @include wider-than($phone-max) {
            display: block;
        }
    }

    .tab-controller {
        width: 100%;

        .tab {
            @include set-font(body-large);

            font-weight: normal;

            @include wider-than($tablet-max) {
                font-size: 150%;
                padding-bottom: $normal-margin;
            }
        }
    }

    .tab-content {
        padding: 4rem 0 6rem;
        width: 100%;
    }

    .unbold {
        font-weight: normal;
    }

    .resources {
        display: grid;
        grid-gap: 2 * $normal-margin  2 * $normal-margin;
        margin: 2rem 0 0;

        @include wider-than($phone-max) {
            grid-template-columns: 1fr 1fr;
        }

        @include wider-than($tablet-max) {
            grid-template-columns: 1fr 1fr 1fr;
        }

        &.ally {
            @include wider-than($phone-max) {
                grid-template-columns: 1fr 1fr 1fr;
            }

            @include wider-than($tablet-max) {
                grid-template-columns: 1fr 1fr 1fr 1fr;
            }
        }
    }

    .btn.default {
        background-color: ui-color(white);
    }

    .table-of-contents {
        list-style: none;
        margin: 0;
        padding: 0;
        padding-left: 2rem;

        > li {
            font-weight: bold;
            margin-bottom: 2rem;
        }

        ol {
            font-weight: normal;
            list-style: none;
            padding-left: 2rem;

            .box {
                font-weight: bold;
            }
        }
    }

}
