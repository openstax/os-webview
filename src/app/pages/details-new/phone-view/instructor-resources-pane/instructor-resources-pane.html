<template args="model">
    <h4>Free ({model.resources.freeResources && model.resources.freeResources.length})</h4>
    <div class="swipe-zone">
        <div class="free-resources-region" skip="true"></div>
    </div>
    <h4>Paid ({model.resources.paidResources && model.resources.paidResources.length})</h4>
    <div class="swipe-zone">
        <div class="paid-resources-region">
            <div class="logo-box" each="res in model.resources.paidResources">
                <a href="{res.book_link_url}">
                    <img src="{res.ally_color_logo}" alt="{res.ally_heading}">
                </a>
            </div>
        </div>
    </div
</template>
