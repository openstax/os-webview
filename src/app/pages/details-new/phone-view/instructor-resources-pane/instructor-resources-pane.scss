@import 'pattern-library/core/pattern-library/headers';

.instructor-resources-pane {
    padding: 0 2rem 2.5rem;

    .swipe-zone {
        // TODO Handle swiping rather than scroll bar
        margin: 0 -0.3rem;
        overflow-x: scroll;
        padding: 0.3rem; // room for the shadow
        width: calc(100vw - #{2 * $normal-margin});
    }

    .free-resources-region,
    .paid-resources-region {
        display: grid;

        h3 {
            @include set-font(h4);

            margin-bottom: 2rem;
        }
    }

    .free-resources-region {
        grid-gap: $normal-margin;
        grid-template-columns: repeat(50, 20rem);
    }

    .paid-resources-region {
        grid-gap: 1rem;
        grid-template-columns: repeat(5, 7.5rem);
    }

    .logo-box {
        align-items: center;
        border-radius: 0.4rem;
        // TODO Replace with variable, probably
        box-shadow: 0 2px 6px 0 rgba(0, 0, 0, 0.2);
        display: flex;
        height: 6.4rem;
        justify-content: center;

        img {
            max-width: 7.5rem;
            padding: $tight-margin;
        }
    }
}
