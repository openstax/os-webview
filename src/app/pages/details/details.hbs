<style scoped>
    @import url(/app/pages/details/details.css);
    @import url(https://fonts.googleapis.com/css?family=Lato:400,900);
</style>
<div class="details-page">
        <section class="container book-info">
            <img class="book-cover" />
                <section class="details">
                    <h1 class="title"></h1>
                    <div class="tabs">
                        <a class="table-of-contents-link" class="tab"><span>Table of Contents</span></a>
                        <div class="table-of-contents hidden">
                            <div class="box"></div>
                        </div>
                        <a href="#resources" class="tab"><span>Instructor Resources</span></a>
                        <a href="#student-resources-section" class="tab"><span>Student Resources</span></a>
                        <a href="#details" class="tab"><span>Details</span></a>
                        <a href="#errata" class="tab errata-tab"><span>Errata</span></a>
                    </div>
                    <p class="blurb"></p>
                    <h2>Senior Contributing Authors</h2>
                    <div id="top-authors"></div>
                    <div class="links">
                        <a class="go-to" href="#details">See all authors</a>
                    </div>
                </section>
                <section class="floating-menu">
                    <div class="box">
                    <!-- <h3>Get this title</h3> -->
                    <div class="get-this-book"></div>
                    <div class="notes">
                        <!-- <small><sup>*</sup>Our website is Section 508 Level A accessible.</small> -->
                    </div>
                    <div class="cta">
                        <p class="interested-text">
                            Using this book, or interested in adopting it for your
                            course?
                        </p>
                        <a href="/adoption?{{title}}" class="btn">Adopt This Book</a>
                        <a href="/interest?{{title}}" class="secondary btn">get interested</a>
                    </div>
                </div>
                </section>
        </section>
        <section class="endorsement">
            <div class="container">
                <blockquote>
                    <div id="endorsement"></div>
                        <p class="attribution">
                                <span>&mdash; <span id="attribution"></span></span>
                                <span id="attribution-school"></span>
                        </p>

                </blockquote>
            </div>
            {{{strips}}}
        </section>
        <section id="resources" class="resources block">
            <div class="container">
            <div class="row">
                <h3>OpenStax Instructor Resources</h3>
                <p>
                    Access these free resources with your verified instructor account.
                    <span id="extra-instructions"></span>
                </p>
                <div id="instructor-resources" class="resource"></div>
            </div>
            <div class="row">
            <h3>Ally Resources</h3>
            <p>
                OpenStax Allies provide additional, low-cost resources for your
                book. See whatâ€™s offered for this title below and learn more about
                our <a href="/allies">Allies here</a>.</p>
            <div id='allies' class="resource"></div>
            </div>
            <div id="student-resources-section" class="row">
                <h3>Student Resources</h3>
                <p>
                    Take a look at our free student resources that are integrated
                    with your book to help you ace your course.
                </p>
                <div id="student-resources" class="resource"></div>
            </div>
            <div class="cta hidden">
                <a class="btn handbook-link">Learn how you can help bring more
                    OpenStax books to your campus!</a>
            </div>
        </div>
        </section>
            <section id="details" class="authors">
                <div class="container">
                <h2>Details</h2>
                <h3>Authors</h3>
                <div id="all-authors"></div>
            </div>
            </section>
            <section id="errata" class="errata">
                <div class="container">
                <h2>Errata</h2>
                <p>
                    All OpenStax textbooks undergo a rigorous review process. However, like
                    any professional-grade textbook, errors sometimes occur. The good part
                    is, since our books are web-based, we can make updates periodically. If
                    you have a correction to suggest, submit it here. We'll review your
                    suggestion and make necessary changes.
                </p>
                <ul class="links">
                    <li><a class="go-to errata-link">Errata list</a></li>
                    <li><a class="go-to suggest-correction-link">Suggest a correction</a></li>
                </ul>
            </div>
            </section>
</div>
