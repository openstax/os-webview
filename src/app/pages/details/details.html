<template args="model">
    <div class="hero {model.reverseGradient ? 'reverse-gradient' : ''}" data-slug="{model.slug}">
        <div class="content book-title">
            <if condition="model.titleImage">
                <h1 class="image-heading">
                    <img class="title-image" src="{model.titleImage}" alt={model.bookTitle}>
                </h1>
            </if>
        </div>
    </div>
    <div class="content-wrapper">
        <div class="content">
            <div class="phone-view" skip="true"></div>
            <div class="bigger-view">
                <div class="tab-controller" skip="true"></div>
                <div class="tab-content" skip="true"></div>
            </div>
        </div>
    </div>
</template>
