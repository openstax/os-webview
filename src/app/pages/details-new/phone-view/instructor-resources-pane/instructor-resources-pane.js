import VERSION from '~/version';
import {Controller} from 'superb.js';
import settings from 'settings';
import ResourceBox from '../../resource-box/resource-box';
import {description as template} from './instructor-resources-pane.html';

export default class InstructorResourcePane extends Controller {

    init(props) {
        this.props = props;
        this.template = template;
        this.view = {
            classes: ['instructor-resources-pane']
        };
        this.regions = {
            freeResources: '.free-resources-region'
        };
        /* eslint max-len: 0 */
        this.css = `/app/pages/details-new/phone-view/instructor-resources-pane/instructor-resources-pane.css?${VERSION}`;
        // Static model
        this.model = {
            resources: props.resources
        };
    }

    onLoaded() {
        this.props.userStatusPromise.then((userStatus) => {
            for (const res of this.props.resources.freeResources) {
                const resourceBox = new ResourceBox(
                    Object.assign({
                        heading: res.resource_heading,
                        description: res.resource_description
                    }, ResourceBox.instructorResourceBoxPermissions(res, userStatus))
                );

                this.regions.freeResources.append(resourceBox);
            }
            // Paid resources are handled by the template
        });
    }

}
