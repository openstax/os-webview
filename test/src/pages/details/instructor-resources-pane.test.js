import {InstructorResourcePane} from '~/pages/details/phone-view/instructor-resources-pane/instructor-resources-pane.jsx';
import details from '../../data/details.js'
import {makeMountRender, snapshotify} from '../../../helpers/jsx-test-utils.jsx';

const instructorResources = details.book_faculty_resources;
const instructor = {
    isStudent: false,
    isInstructor: true,
    userInfo: {
        id: 2
    }
};

describe('InstructorResourcePane (as instructor)', () => {
    it('matches snapshot', () => {
        const wrapper = makeMountRender(InstructorResourcePane, {
            bookId: 42,
            compCopyDialogProps: {},
            featuredResources: [],
            videoResources: [],
            blogLinkResources: [],
            referenceResources: [],
            otherResources: instructorResources,
            userStatusPromise: Promise.resolve(instructor)
        })();

        expect(snapshotify(wrapper)).toMatchSnapshot();
    });
});
