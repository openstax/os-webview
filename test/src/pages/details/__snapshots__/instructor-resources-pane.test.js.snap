// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InstructorResourcePane (as instructor) matches snapshot 1`] = `"<div class=\\"instructor-resources-pane\\"><a class=\\"card filter-for-book\\">OpenStax Partners <svg aria-hidden=\\"true\\" focusable=\\"false\\" data-prefix=\\"fas\\" data-icon=\\"sign-out-alt\\" class=\\"svg-inline--fa fa-sign-out-alt fa-w-16 \\" role=\\"img\\" xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 512 512\\"><path fill=\\"currentColor\\" d=\\"M497 273L329 441c-15 15-41 4.5-41-17v-96H152c-13.3 0-24-10.7-24-24v-96c0-13.3 10.7-24 24-24h136V88c0-21.4 25.9-32 41-17l168 168c9.3 9.4 9.3 24.6 0 34zM192 436v-40c0-6.6-5.4-12-12-12H96c-17.7 0-32-14.3-32-32V160c0-17.7 14.3-32 32-32h84c6.6 0 12-5.4 12-12V76c0-6.6-5.4-12-12-12H96c-53 0-96 43-96 96v192c0 53 43 96 96 96h84c6.6 0 12-5.4 12-12z\\"></path></svg></a><div class=\\"free-resources-region\\"><a class=\\"resource-box\\" href=\\"https://d3bxy9euw4e147.cloudfront.net/oscms-dev/media/documents/OpenStax_Instructor_Getting_Started_Guide_Fall_2018.pdf\\" data-local=\\"false\\"><div class=\\"reference-number\\">1</div><div class=\\"top\\"><div class=\\"top-line\\"><h3 class=\\"\\">Instructor Getting Started Guide</h3></div><div class=\\"description\\"></div></div><div class=\\"bottom\\"><div class=\\"left\\">Download</div><div class=\\"right\\"><svg aria-hidden=\\"true\\" focusable=\\"false\\" data-prefix=\\"fas\\" data-icon=\\"download\\" class=\\"svg-inline--fa fa-download fa-w-16 \\" role=\\"img\\" xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 512 512\\"><path fill=\\"currentColor\\" d=\\"M216 0h80c13.3 0 24 10.7 24 24v168h87.7c17.8 0 26.7 21.5 14.1 34.1L269.7 378.3c-7.5 7.5-19.8 7.5-27.3 0L90.1 226.1c-12.6-12.6-3.7-34.1 14.1-34.1H192V24c0-13.3 10.7-24 24-24zm296 376v112c0 13.3-10.7 24-24 24H24c-13.3 0-24-10.7-24-24V376c0-13.3 10.7-24 24-24h146.7l49 49c20.1 20.1 52.5 20.1 72.6 0l49-49H488c13.3 0 24 10.7 24 24zm-124 88c0-11-9-20-20-20s-20 9-20 20 9 20 20 20 20-9 20-20zm64 0c0-11-9-20-20-20s-20 9-20 20 9 20 20 20 20-9 20-20z\\"></path></svg></div></div></a><a class=\\"resource-box\\" href=\\"https://cms-dev.openstax.org/oxauth/login/?next=https%3A%2F%2Fcms-dev.openstax.org%2F%3FInstructor%20resources\\" data-local=\\"true\\"><div class=\\"reference-number\\">2</div><div class=\\"top\\"><div class=\\"top-line\\"><h3 class=\\"\\">PowerPoint Slides</h3></div><div class=\\"description\\"></div></div><div class=\\"bottom\\"><div class=\\"left\\">Log in to unlock</div><div class=\\"right\\"><svg aria-hidden=\\"true\\" focusable=\\"false\\" data-prefix=\\"fas\\" data-icon=\\"lock\\" class=\\"svg-inline--fa fa-lock fa-w-14 \\" role=\\"img\\" xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 448 512\\"><path fill=\\"currentColor\\" d=\\"M400 224h-24v-72C376 68.2 307.8 0 224 0S72 68.2 72 152v72H48c-26.5 0-48 21.5-48 48v192c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V272c0-26.5-21.5-48-48-48zm-104 0H152v-72c0-39.7 32.3-72 72-72s72 32.3 72 72v72z\\"></path></svg></div></div></a><a class=\\"resource-box\\" href=\\"https://cms-dev.openstax.org/oxauth/login/?next=https%3A%2F%2Fcms-dev.openstax.org%2F%3FInstructor%20resources\\" data-local=\\"true\\"><div class=\\"top\\"><div class=\\"top-line\\"><h3 class=\\"\\">Instructor Solutions Manual</h3></div><div class=\\"description\\"></div></div><div class=\\"bottom\\"><div class=\\"left\\">Log in to unlock</div><div class=\\"right\\"><svg aria-hidden=\\"true\\" focusable=\\"false\\" data-prefix=\\"fas\\" data-icon=\\"lock\\" class=\\"svg-inline--fa fa-lock fa-w-14 \\" role=\\"img\\" xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 448 512\\"><path fill=\\"currentColor\\" d=\\"M400 224h-24v-72C376 68.2 307.8 0 224 0S72 68.2 72 152v72H48c-26.5 0-48 21.5-48 48v192c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V272c0-26.5-21.5-48-48-48zm-104 0H152v-72c0-39.7 32.3-72 72-72s72 32.3 72 72v72z\\"></path></svg></div></div></a><a class=\\"resource-box\\" href=\\"https://d3bxy9euw4e147.cloudfront.net/oscms-dev/media/documents/Syllabus_Language_-_College_Algebra_NUFM3pN.txt\\" data-local=\\"false\\"><div class=\\"top\\"><div class=\\"top-line\\"><h3 class=\\"\\">Sample Syllabus Language</h3></div><div class=\\"description\\"></div></div><div class=\\"bottom\\"><div class=\\"left\\">Download</div><div class=\\"right\\"><svg aria-hidden=\\"true\\" focusable=\\"false\\" data-prefix=\\"fas\\" data-icon=\\"download\\" class=\\"svg-inline--fa fa-download fa-w-16 \\" role=\\"img\\" xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 512 512\\"><path fill=\\"currentColor\\" d=\\"M216 0h80c13.3 0 24 10.7 24 24v168h87.7c17.8 0 26.7 21.5 14.1 34.1L269.7 378.3c-7.5 7.5-19.8 7.5-27.3 0L90.1 226.1c-12.6-12.6-3.7-34.1 14.1-34.1H192V24c0-13.3 10.7-24 24-24zm296 376v112c0 13.3-10.7 24-24 24H24c-13.3 0-24-10.7-24-24V376c0-13.3 10.7-24 24-24h146.7l49 49c20.1 20.1 52.5 20.1 72.6 0l49-49H488c13.3 0 24 10.7 24 24zm-124 88c0-11-9-20-20-20s-20 9-20 20 9 20 20 20 20-9 20-20zm64 0c0-11-9-20-20-20s-20 9-20 20 9 20 20 20 20-9 20-20z\\"></path></svg></div></div></a><a class=\\"resource-box\\" href=\\"https://cms-dev.openstax.org/oxauth/login/?next=https%3A%2F%2Fcms-dev.openstax.org%2F%3FInstructor%20resources\\" data-local=\\"true\\"><div class=\\"top\\"><div class=\\"top-line\\"><h3 class=\\"\\">Canvas Course Cartridge</h3></div><div class=\\"description\\"></div></div><div class=\\"bottom\\"><div class=\\"left\\">Log in to unlock</div><div class=\\"right\\"><svg aria-hidden=\\"true\\" focusable=\\"false\\" data-prefix=\\"fas\\" data-icon=\\"lock\\" class=\\"svg-inline--fa fa-lock fa-w-14 \\" role=\\"img\\" xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 448 512\\"><path fill=\\"currentColor\\" d=\\"M400 224h-24v-72C376 68.2 307.8 0 224 0S72 68.2 72 152v72H48c-26.5 0-48 21.5-48 48v192c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V272c0-26.5-21.5-48-48-48zm-104 0H152v-72c0-39.7 32.3-72 72-72s72 32.3 72 72v72z\\"></path></svg></div></div></a><a class=\\"resource-box\\" href=\\"https://d3bxy9euw4e147.cloudfront.net/oscms-dev/media/documents/College_Algebra_Release_Notes_2017.pdf\\" data-local=\\"false\\"><div class=\\"top\\"><div class=\\"top-line\\"><h3 class=\\"\\">Errata Release Notes</h3></div><div class=\\"description\\"></div></div><div class=\\"bottom\\"><div class=\\"left\\">Download</div><div class=\\"right\\"><svg aria-hidden=\\"true\\" focusable=\\"false\\" data-prefix=\\"fas\\" data-icon=\\"download\\" class=\\"svg-inline--fa fa-download fa-w-16 \\" role=\\"img\\" xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 512 512\\"><path fill=\\"currentColor\\" d=\\"M216 0h80c13.3 0 24 10.7 24 24v168h87.7c17.8 0 26.7 21.5 14.1 34.1L269.7 378.3c-7.5 7.5-19.8 7.5-27.3 0L90.1 226.1c-12.6-12.6-3.7-34.1 14.1-34.1H192V24c0-13.3 10.7-24 24-24zm296 376v112c0 13.3-10.7 24-24 24H24c-13.3 0-24-10.7-24-24V376c0-13.3 10.7-24 24-24h146.7l49 49c20.1 20.1 52.5 20.1 72.6 0l49-49H488c13.3 0 24 10.7 24 24zm-124 88c0-11-9-20-20-20s-20 9-20 20 9 20 20 20 20-9 20-20zm64 0c0-11-9-20-20-20s-20 9-20 20 9 20 20 20 20-9 20-20z\\"></path></svg></div></div></a><a class=\\"resource-box\\" href=\\"https://d3bxy9euw4e147.cloudfront.net/oscms-dev/media/documents/OpenStax_Representation_and_Diversity_Development_Guidelines_2bb08eB.pdf\\" data-local=\\"false\\"><div class=\\"top\\"><div class=\\"top-line\\"><h3 class=\\"\\">Diversity and Representation Development Guidelines</h3></div><div class=\\"description\\"></div></div><div class=\\"bottom\\"><div class=\\"left\\">Download</div><div class=\\"right\\"><svg aria-hidden=\\"true\\" focusable=\\"false\\" data-prefix=\\"fas\\" data-icon=\\"download\\" class=\\"svg-inline--fa fa-download fa-w-16 \\" role=\\"img\\" xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 512 512\\"><path fill=\\"currentColor\\" d=\\"M216 0h80c13.3 0 24 10.7 24 24v168h87.7c17.8 0 26.7 21.5 14.1 34.1L269.7 378.3c-7.5 7.5-19.8 7.5-27.3 0L90.1 226.1c-12.6-12.6-3.7-34.1 14.1-34.1H192V24c0-13.3 10.7-24 24-24zm296 376v112c0 13.3-10.7 24-24 24H24c-13.3 0-24-10.7-24-24V376c0-13.3 10.7-24 24-24h146.7l49 49c20.1 20.1 52.5 20.1 72.6 0l49-49H488c13.3 0 24 10.7 24 24zm-124 88c0-11-9-20-20-20s-20 9-20 20 9 20 20 20 20-9 20-20zm64 0c0-11-9-20-20-20s-20 9-20 20 9 20 20 20 20-9 20-20z\\"></path></svg></div></div></a><a class=\\"resource-box\\" data-local=\\"false\\"><div class=\\"top\\"><div class=\\"top-line\\"><h3 class=\\"\\">Sample Syllabus Language</h3></div><div class=\\"description\\"></div></div><div class=\\"bottom\\"><div class=\\"left\\">View resources</div><div class=\\"right\\"><svg aria-hidden=\\"true\\" focusable=\\"false\\" data-prefix=\\"fas\\" data-icon=\\"download\\" class=\\"svg-inline--fa fa-download fa-w-16 \\" role=\\"img\\" xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 512 512\\"><path fill=\\"currentColor\\" d=\\"M216 0h80c13.3 0 24 10.7 24 24v168h87.7c17.8 0 26.7 21.5 14.1 34.1L269.7 378.3c-7.5 7.5-19.8 7.5-27.3 0L90.1 226.1c-12.6-12.6-3.7-34.1 14.1-34.1H192V24c0-13.3 10.7-24 24-24zm296 376v112c0 13.3-10.7 24-24 24H24c-13.3 0-24-10.7-24-24V376c0-13.3 10.7-24 24-24h146.7l49 49c20.1 20.1 52.5 20.1 72.6 0l49-49H488c13.3 0 24 10.7 24 24zm-124 88c0-11-9-20-20-20s-20 9-20 20 9 20 20 20 20-9 20-20zm64 0c0-11-9-20-20-20s-20 9-20 20 9 20 20 20 20-9 20-20z\\"></path></svg></div></div></a></div></div>"`;
